rllib/core/optim/examples/dummy_ppo_optimizer.py

```diff
@@ -0,0 +1,144 @@
+            "critic_loss": critic_loss,
+        }
+        return loss_dict
+
+    def _configure_optimizers(self):
+        return [
+            torch.optim.SGD(self.module.parameters(), lr=self._config.get("lr", 1e-3))
+        ]
+
+    def get_state(self):
+        return [optim.state_dict() for optim in self.optimizers]
+
+    def set_state(self, state):
+        assert len(state) == len(self.optimizers)
+        for optim, optim_state in zip(self.optimizers, state):
+            optim.load_state_dict(optim_state)
+
+
+class DummyPPOTorchTrainer:
+    def __init__(self, config) -> None:
+        module_config = config["module_config"]
+        optimizer_config = {}
+        self._module = SimplePPOModule(module_config)
+        self._rl_optimizer = DummyPPOTorchOptimizer(self._module, optimizer_config)
+
+    @staticmethod
+    def on_after_compute_gradients(gradients_dict):
+        return gradients_dict
+
+    def compile_results(
+        self, batch, fwd_out, postprocessed_loss, post_processed_gradients
+    ):
+        """Compile results from the update."""
+
+        loss_numpy = convert_to_numpy(postprocessed_loss)
+        results = {
+            "avg_reward": batch["rewards"].mean(),
+        }
+        results.update(loss_numpy)
+        return results
+
+    def update(self, batch):
+        """Perform an update on this Trainer."""
+        torch_batch = convert_to_torch_tensor(batch)
+        fwd_out = self._module.forward_train(torch_batch)
+        loss = self._rl_optimizer.compute_loss(torch_batch, fwd_out)
+        postprocessed_loss = self._rl_optimizer.on_after_compute_loss(loss)
+        gradients = self.compute_gradients(loss)
+        post_processed_gradients = self.on_after_compute_gradients(gradients)
+        self.apply_gradients(post_processed_gradients)
+        results = self.compile_results(
+            batch, fwd_out, postprocessed_loss, post_processed_gradients
+        )
+        return results
+
+    def compute_gradients(self, optimization_vars, **kwargs):
+        """Perform an update on self._module
+
+            For example compute and apply gradients to self._module if
+                necessary.
+
+        Args:
+            optimization_vars: named variables used for optimizing self._module
+
+        Returns:
+            A dictionary of extra information and statistics.
+        """
+        # for torch
+        optimization_vars["total_loss"].backward()
+        grads = {n: p.grad for n, p in self._module.named_parameters()}
+        return grads
+
+    def apply_gradients(self, gradients):
+        """Perform an update on self._module"""
+        del gradients
+        for optimizer in self._rl_optimizer.get_optimizers():
+            optimizer.step()
+            optimizer.zero_grad()
+
+    def set_state(self, state):
+        """Set the state of the trainer."""
+        self._module.set_state(state.get("module_state", {}))
+
+
+if __name__ == "__main__":
```

Comment:
Can you move this main() script to a unittest? Here is my suggestion: use offline expert data from cartpole. Create a dummy trainer for behavioral cloning. Then test the following: 
1) Does loss actually go down?
2) When you do inference on the learned agent does it achieve a higher reward than the initial baseline? This can also be an absolute test for reaching a minimum reward.
3) Test a few important loss postprocessing and gradient postprocessings that we do in rllib today and see if they work here. 
This exercise not only will become a good unittest, it will expose the downsides of the designed API (if any)